vscode/microsoft-kiota/src/utilities/deep-linking.ts

```diff
@@ -28,6 +29,7 @@ export function transformToGenerationConfig(deepLinkParams: Partial<IntegrationP
         break;
       case "apimanifest":
         generationConfig.generationType = "apimanifest";
+        generationConfig.pluginTypes = ["ApiManifest"];
```

Comment:
Maybe try the following:
`
switch (deepLinkParams.type) {
  case "apiplugin":
    generationConfig.generationType = "plugin";
    generationConfig.pluginTypes = ["ApiPlugin"];
    break;
  case "openai":
    generationConfig.generationType = "other";
    generationConfig.pluginTypes = ['OpenAI'];
    break;
  case "apimanifest":
    generationConfig.generationType = "other";
    generationConfig.pluginTypes = ["ApiManifest"];
    break;
 }
`
followed by suggested changes on `chooseOtherGenerationType`

Reply:
This sets the generationType for apimanifest and openai to other for starters, then later in the generatesteps functions, changes this values to the expected generation steps.